homeassistant/components/fireservicerota/config_flow.py

```diff
@@ -0,0 +1,129 @@
+"""Config flow for FireServiceRota."""
+from pyfireservicerota import FireServiceRota, InvalidAuthError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_TOKEN, CONF_URL, CONF_USERNAME
+
+from .const import DOMAIN, URL_LIST  # pylint: disable=unused-import
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_URL, default="www.brandweerrooster.nl"): vol.In(URL_LIST),
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+class FireServiceRotaFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a FireServiceRota config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Initialize config flow."""
+        self.api = None
+        self._base_url = None
+        self._username = None
+        self._password = None
+        self._existing_entry = None
+        self._description_placeholders = None
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        errors = {}
+
+        if user_input is None:
+            return self._show_setup_form(user_input, errors)
+
+        return await self._validate_and_create_entry(user_input, "user")
+
+    async def _validate_and_create_entry(self, user_input, step_id):
+        """Check if config is valid and create entry if so."""
+        self._password = user_input[CONF_PASSWORD]
+
+        extra_inputs = user_input
+
+        if self._existing_entry:
+            extra_inputs = self._existing_entry
```

Comment:
We seem to be missing test cases in the config flow. It needs 100% test coverage.